src/chatdMsg.h

```diff
@@ -789,6 +779,35 @@ class Message: public Buffer
         return (type == kMsgContainsMeta && dataSize() > 3) ? std::string(buf()+3, dataSize() - 3) : "";
     }
 
+    bool isMissingCall(karere::Id myHandle) const
+    {
+        if (type != kMsgCallEnd)
+        {
+            return false;
+        }
+
+        uint8_t termCode = Message::extractTermCodeEndCall(*this);
+
+        return !isOwnMessage(myHandle) && (termCode == CallDataReason::kNoAnswer || termCode == CallDataReason::kCancelled);
+    }
+
+    static uint8_t extractTermCodeEndCall(const Buffer& buffer)
+    {
+        unsigned int lenCallid = sizeof (karere::Id);
+        unsigned int lenDuration = sizeof (uint32_t);
+        unsigned int lenTermCode = sizeof (uint8_t);
+
+        if (buffer.size() < (lenCallid + lenDuration + lenTermCode))
+        {
+            return -1;
```

Comment:
This method returns unsigned int
```suggestion
            return 0xFF;
```